server/controllers/projectController.js

```diff
@@ -0,0 +1,59 @@
+const Project = require("../models/projects");
+const User = require("../models/User");
+
+async function getAllProjects(req, res) {
+  const projects = await Project.find().sort({ name: 1 });
+  if (!projects) {
+    return res.status(404);
+  }
+  res.send(projects);
+}
+
+async function getProjectsForUser(req, res) {
+  const projects = await User.find({ _id: req.params.userId })
+    .select({ username: 1 })
+    .populate("projects", "title");
+
+  if (!projects) {
+    return res.status(404);
+  }
+  res.send(projects);
+}
+
+async function createProjectForUser(req, res) {
+  try {
+    // Get authorized user
+    const user = await User.findById(req.id).select("-password");
+    if (!user) {
+      return res.status(404).send("User not found");
+    }
```

Comment:
Will we need to do this check all the time? If so we might want to do this in the middleware in case we are concerned users are deleted. We know that this user once existed, because it passed the JWT check in the middleware. 

If we are not concerned about this, then we maybe don't need to make this extra database call to get the user, and can just update it later on as we have confirmed the user was once existed and authenticated from the JWT middleware check

Reply:
Removed the database call to User every time project is created.